avr-hal-generic/src/port.rs

```diff
@@ -257,6 +281,66 @@ macro_rules! impl_generic_pin {
                     })
                 }
             }
+
+            impl digital::OutputPin for $GenericPin<mode::OpenDrain> {
+                type Error = Void;
+
+                fn set_high(&mut self) -> Result<(), Self::Error> {
+                    match self {
+                        $(
+                            $GenericPin::$PortEnum(i, _) => unsafe {
+                                (*<$PORTX>::ptr())
+                                    .$reg_ddr
+                                    .modify(|r, w| {
+                                        w.bits(r.bits() & !(1 << *i))
+                                    })
+                            },
+                        )+
+                    }
+                    Ok(())
+                }
+
+                fn set_low(&mut self) -> Result<(), Self::Error> {
+                    match self {
+                        $(
+                            $GenericPin::$PortEnum(i, _) => unsafe {
+                                (*<$PORTX>::ptr())
+                                    .$reg_ddr
+                                    .modify(|r, w| {
+                                        w.bits(r.bits() | (1 << *i))
+                                    })
```

Comment:
Minor nit: Can you match the style here with what you did further down?  I'd think it is a bit more readable as it is more compact:

```rust
(*<$PORTX>::ptr()).$reg_port.modify(|r, w| {
    w.bits(r.bits() & !(1 << $i))
});
```

Reply:
I will fix that.